src/neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -155,30 +143,35 @@ private bool UnregisterCandidate(ApplicationEngine engine, ECPoint pubkey)
         private bool Vote(ApplicationEngine engine, UInt160 account, ECPoint voteTo)
         {
             if (!engine.CheckWitnessInternal(account)) return false;
-            return VoteInternal(engine.Snapshot, account, voteTo);
-        }
-
-        private bool VoteInternal(StoreView snapshot, UInt160 account, ECPoint voteTo)
-        {
             StorageKey key_account = CreateStorageKey(Prefix_Account).Add(account);
-            if (snapshot.Storages.TryGet(key_account) is null) return false;
-            StorageItem storage_account = snapshot.Storages.GetAndChange(key_account);
+            if (engine.Snapshot.Storages.TryGet(key_account) is null) return false;
+            StorageItem storage_account = engine.Snapshot.Storages.GetAndChange(key_account);
             NeoAccountState state_account = storage_account.GetInteroperable<NeoAccountState>();
+            if (state_account.VoteTo is null ^ voteTo is null)
```

Comment:
Why we check here `voteTo`?

Reply:
This is the case that voters count changes.

Reply:
If voteTo it's null, do you think that we should increase `votersCount`?

Reply:
In fact, it's a vote, but if all people vote null and we have only one vote, what should be the committee?

Reply:
> If voteTo it's null, do you think that we should increase votersCount?

If `voteTo` is `null`, we will decrease `votersCount`.

https://github.com/neo-project/neo/blob/42efe85b6b8c030c151e4ead7f06dec12115fb11/src/neo/SmartContract/Native/Tokens/NeoToken.cs#L157

Reply:
> In fact, it's a vote, but if all people vote null and we have only one vote, what should be the committee?

Then we use the default one.

https://github.com/neo-project/neo/blob/42efe85b6b8c030c151e4ead7f06dec12115fb11/src/neo/SmartContract/Native/Tokens/NeoToken.cs#L211-L221